test/functional/legacy/013_autocommands_on_close_spec.lua

```diff
@@ -0,0 +1,112 @@
+-- Tests for autocommands on :close command
+--
+-- Write three files and open them, each in a window.
+-- Then go to next window, with autocommand that deletes the previous one.
+-- Do this twice, writing the file.
+--
+-- Also test deleting the buffer on a Unload event.  If this goes wrong there
+-- will be the ATTENTION prompt.
+--
+-- Also test changing buffers in a BufDel autocommand.  If this goes wrong there
+-- are ml_line errors and/or a Crash.
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('autocommands on close', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([=[
+      start of testfile
+      	contents
+      	contents
+      	contents
+      end of testfile]=])
+
+    -- Write the current buffer to "test13.in", else subsequent
+    -- steps will fail
+    execute('w! test13.in')
+    execute(':/^start of testfile/,/^end of testfile/w! Xtestje1')
+    execute(':/^start of testfile/,/^end of testfile/w! Xtestje2')
+    execute(':/^start of testfile/,/^end of testfile/w! Xtestje3')
+    execute('e Xtestje1')
```

Comment:
Is this really needed? Can't you just do it like this:
```lua
execute('w! Xtestje1')
execute('w! Xtestje2')
execute('w! Xtestje3')
```
The first `w` of an unnamed buffer should set the file so that you don't even need the `e Xtestje1`.